src/alt/store/index.js

```diff
@@ -53,7 +53,13 @@ export function createStoreConfig(globalConfig, StoreModel) {
         return state
       }
     },
-    setState: fn.assign
+    setState(currentState, nextState) {
+      if (Object.prototype.toString.call(nextState) === '[object Object]') {
```

Comment:
With something like #344 it could be a non-issue as you could potentially have a separate overriding implementation of `getState` for stores holding immutable data.